db/model/profile.js

```diff
@@ -124,6 +124,16 @@ module.exports = function profile(seq, dataTypes) {
           .catch((err) => reject(err));
         });
       }, // isAdmin
+
+      hasWriteAccess(profileId, modelName) {
+        const accessModel = modelName.charAt(0).toLowerCase() + modelName.slice(1) + 'Access';
```

Comment:
Better to create a mapping of the model to the profile access field name than to rely on naming convention... maybe add a static function to each of the models which returns its profile access column name?